hooks/pre/20-certs_update.rb

```diff
@@ -0,0 +1,47 @@
+require 'fileutils'
+require 'English'
+
+if module_enabled?('certs')
+  SSL_BUILD_DIR = param('certs', 'ssl_build_dir').value
+
+  def mark_for_update(cert_name, hostname = nil)
+    path = File.join(*[SSL_BUILD_DIR, hostname, cert_name].compact)
+    if app_value(:noop)
+      puts "Marking certificate #{path} for update (noop)"
+    else
+      puts "Marking certificate #{path} for update"
+      FileUtils.touch("#{path}.update")
+    end
+  end
+
+  if param('foreman_proxy_certs', 'foreman_proxy_fqdn')
+    hostname = param('foreman_proxy_certs', 'foreman_proxy_fqdn').value
+  else
+    hostname = param('certs', 'node_fqdn').value
+  end
+
+  if app_value('certs_update_server')
+    mark_for_update("#{hostname}-apache", hostname)
+    mark_for_update("#{hostname}-foreman-proxy", hostname)
+  end
+
+  if app_value('certs_update_all') || app_value('certs_update_default_ca') || app_value('certs_reset')
+    all_cert_names = Dir.glob(File.join(SSL_BUILD_DIR, hostname, '*.noarch.rpm')).map do |rpm|
+      File.basename(rpm).sub(/-1\.0-\d+\.noarch\.rpm/, '')
+    end.uniq
+
+    all_cert_names.each do |cert_name|
+      mark_for_update(cert_name, hostname)
+    end
+  end
+
+  if app_value('certs_update_server_ca') || app_value('certs_reset')
+    mark_for_update('katello-server-ca')
+  end
```

Comment:
Does it make sense to gather an array on all of this? We now have `all_cert_names` but really we can gather them all. Explicitly gathering the paths would make `mark_for_update` redundant. Because I wanted to know how it looked, I took a stab at it:
```ruby
require 'fileutils'

if module_enabled?('certs')
  if param('foreman_proxy_certs', 'foreman_proxy_fqdn')
    hostname = param('foreman_proxy_certs', 'foreman_proxy_fqdn').value
  else
    hostname = param('certs', 'node_fqdn').value
  end

  SSL_BUILD_DIR = param('certs', 'ssl_build_dir').value
  HOST_BUILD_DIR = File.join(SSL_BUILD_DIR, hostname)

  certs_to_update = []

  if app_value('certs_update_server')
    certs_to_update << File.join(HOST_BUILD_DIR, "#{hostname}-apache")
    certs_to_update << File.join(HOST_BUILD_DIR, "#{hostname}-foreman-proxy")
  end
   
  if app_value('certs_update_all') || app_value('certs_update_default_ca') || app_value('certs_reset')
    certs_to_update += Dir.glob(File.join(HOST_BUILD_DIR, '*.noarch.rpm')).map do |rpm|
      rpm.sub(/-1\.0-\d+\.noarch\.rpm/, '')
    end
  end

  if app_value('certs_update_server_ca') || app_value('certs_reset')
    certs_to_update << File.join(SSL_BUILD_DIR, 'katello-server-ca')
  end

  certs_to_update.uniq.each do |path|
    if app_value(:noop)
      puts "Marking certificate #{path} for update (noop)"
    else
      puts "Marking certificate #{path} for update"
      FileUtils.touch("#{path}.update")
    end
  end
end
```

I think it looks and better shows the intention of this hook. Note I left out resetting the params since I believe that should be in a different hook (https://github.com/theforeman/foreman-installer/pull/514/files#r476563297).